pkg/glusterfs/controllerserver.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2018 The Gluster CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package glusterfs
+
+import (
+	"fmt"
+
+	dstrings "strings"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/golang/glog"
+	gcli "github.com/heketi/heketi/client/api/go-client"
+	gapi "github.com/heketi/heketi/pkg/glusterfs/api"
+	"github.com/kubernetes-csi/drivers/pkg/csi-common"
+	"github.com/pborman/uuid"
+	"golang.org/x/net/context"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+const (
+	defaultGlusterURL    = "http://127.0.0.1:8081"
+	defaultGlusterServer = "127.0.0.1"
+)
+
+type controllerServer struct {
+	*csicommon.DefaultControllerServer
+}
+
+func GetVersionString(ver *csi.Version) string {
+	return fmt.Sprintf("%d.%d.%d", ver.Major, ver.Minor, ver.Patch)
+}
+
+func (cs *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
+
+	if err := cs.Driver.ValidateControllerServiceRequest(req.Version, csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME); err != nil {
+		glog.V(3).Infof("invalid create volume req: %v", req)
+		return nil, err
+	}
+
+	// Volume Size - Default is 1 GiB
+	volSizeBytes := int64(1 * 1024 * 1024 * 1024)
+	if req.GetCapacityRange() != nil {
+		volSizeBytes = int64(req.GetCapacityRange().GetRequiredBytes())
+	}
+	volSizeGB := int(volume.RoundUpSize(volSizeBytes, 1024*1024*1024))
+
+	//Volume Name
+	volName := req.GetName()
+	if len(volName) == 0 {
+		volName = uuid.NewUUID().String()
+	}
+
+	// Volume Parameters
+	glusterVol := req.GetParameters()["glustervol"]
+	glusterServer := req.GetParameters()["glusterserver"]
+	glusterURL := req.GetParameters()["glusterurl"]
+	glusterURLPort := req.GetParameters()["glusterurlport"]
+	glusterUser := req.GetParameters()["glusteruser"]
+	glusterUserSecret := req.GetParameters()["glusterusersecret"]
+
+	if len(glusterServer) == 0 {
+		glusterServer = defaultGlusterServer
+	}
+
+	if len(glusterURL) == 0 {
+		glusterURL = "http://127.0.0.1:8081"
+	} else {
+		glusterURL = glusterURL + ":" + glusterURLPort
+	}
```

Comment:
This above block is a little strange imho. 
What is the relationship of the input parameters `glusterURL`, `glusterServer` and `glusterURLPort` ?

I would expect s/th like this:

```go
if len(glusterServer) == 0 {
    glusterServer = default
}
if len(glusterURLPort) == 0 {
    glusterURLPort = defaultGlusterURLPort
}
if len(glusterURL) == 0 {
   glusterURL = "http://" + glusterServer + ":" + glusterURLPort
}
```
but maybe i just misunderstand the intentions.

Also why check for `len(string) == 0` instead of `string == ""` ?